internal/policy/policy.go

```diff
@@ -765,3 +765,48 @@ func (s *State) findDelegationEntry(roleName string) (*tuf.Delegation, error) {
 		}
 	}
 }
+
+func ListRules(ctx context.Context, repo *git.Repository) ([]tuf.Delegation, error) {
+	state, err := LoadCurrentState(ctx, repo)
+	if err != nil {
+		return nil, err
+	}
+
+	topLevelTargetsMetadata, err := state.GetTargetsMetadata(TargetsRoleName)
+	if err != nil {
+		return nil, err
+	}
+
+	delegationTargetsMetadata := map[string]*tuf.TargetsMetadata{}
+	for name, env := range state.DelegationEnvelopes {
+		targetsMetadata := &tuf.TargetsMetadata{}
+
+		envBytes, err := env.DecodeB64Payload()
+		if err != nil {
+			return nil, err
+		}
+
+		if err := json.Unmarshal(envBytes, targetsMetadata); err != nil {
+			return nil, err
+		}
+		delegationTargetsMetadata[name] = targetsMetadata
+	}
+
+	delegationsToSearch := topLevelTargetsMetadata.Delegations.Roles
+
+	allDelegations := topLevelTargetsMetadata.Delegations.Roles
+
+	for len(delegationsToSearch) > 0 {
+		delegation := delegationsToSearch[0]
+		delegationsToSearch = delegationsToSearch[1:]
+
+		if state.HasTargetsRole(delegation.Name) {
+			// TODO: clarifying terminating / reachability
+
+			allDelegations = append(allDelegations, delegationTargetsMetadata[delegation.Name].Delegations.Roles...)
```

Comment:
The order matters here. I suggest placing the delegations upfront so that delegated rules are placed right after the delegating rule. I also wonder if it'd be possible indent per delegation level in the CLI output so it's immediately clear how a rule is reached. WDYT?

Reply:
I have changed the ListRules func, so that it returns a slice of delegations with their depth, so different delegations, based on their depth in the delegation tree are indented differently.